src/coreclr/debug/createdump/crashinfomac.cpp

```diff
@@ -277,6 +277,9 @@ void CrashInfo::VisitSegment(MachOModule& module, const segment_command_64& segm
             uint64_t start = segment.vmaddr + module.LoadBias();
             uint64_t end = start + segment.vmsize;
 
+            // Add this module segment to the set used by the thread unwinding to lookup the module base address for an ip.
+            AddModuleAddressRange(start, end, module.BaseAddress());
```

Comment:
Doesn't this mean we'll end up adding this modules base address potentially many times (it will not add it since it's a set, but feels unnecessary). Where are we adding it now that it wasn't before? I'd expect that if moduleMappings has the range, we've seen the base mapping and we'll have stored it already. Or was the masking causing issues for lookups?

Reply:
The module segments are only visited once so there really shouldn't be dups (and like you said it is a set that the range is being added to so it won't be a problem even if there are dups). This to fix cases where the text segment wasn't being added causing the unwind lookup to fail.

The problem was the "find"/add to m_moduleMappings.  There are a lot of times where the "overlapping" warning path is taken. I'm not sure exactly why and didn't want to make anymore changes than necessary.  Later in 7.0, I'll look into this some more.

Reply:
The start/end doesn't need to be rounded to page boundaries for the module address lookup set. The rounding may explain why their are overlapping warning in the module mappings set.